src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -0,0 +1,371 @@
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.IdeSuggestion,
+                                                                         description: s_localizableDescriptionCA2253,
+                                                                         isPortedFxCopRule: true,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: true);
+
+        internal static DiagnosticDescriptor CA2254Rule = DiagnosticDescriptorHelper.Create(CA2254RuleId,
+                                                                         s_localizableTitleCA2254,
+                                                                         s_localizableMessageCA2254,
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.BuildWarning,
+                                                                         description: s_localizableDescriptionCA2254,
+                                                                         isPortedFxCopRule: true,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(CA1727Rule, CA1848Rule, CA2252Rule, CA2253Rule, CA2254Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.EnableConcurrentExecution();
+
+            context.RegisterCompilationStartAction(analysisContext =>
+            {
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(analysisContext.Compilation);
+
+                if (!wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerExtensions, out var loggerExtensionsType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingILogger, out var loggerType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerMessage, out var loggerMessageType))
+                {
+                    return;
+                }
+
+                analysisContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeInvocation(syntaxContext, loggerType, loggerExtensionsType, loggerMessageType), SyntaxKind.InvocationExpression);
```

Comment:
Rename `syntaxContext` to `context`

Reply:
Any reason not to do operation analysis instead of syntax analysis?
operation is almost always preferred if it's possible.

Reply:
Currently working with @sharwell to get this resolved (and use operation analysis instead of syntax analysis).